wpilibc/src/main/native/include/frc/interfaces/Gyro3D.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <frc/interfaces/Gyro.h>
+
+#include <units/angle.h>
+
+#include "frc/geometry/Rotation3d.h"
+
+namespace frc {
+
+/** Gyro that supports representation in the 3D space. */
+class Gyro3D : virtual public Gyro {
```

Comment:
It doesn't make sense for Gyro3D to implement Gyro because it's not designed to be a 2D subinterface.

Reply:
My thought process was how the Pose3d class supports getting a Pose2d by removing the 3rd axis. This makes it so users using vendor code don't experience breaking changes.

Reply:
Breaking interface changes are the point of having a new base class. There's improvements to be made.

Reply:
That I agree with, what changes would you want to add?

Reply:
The interface would be similar to ADIS16470_IMU. There would be roll, pitch, and yaw getters that return units::degree_t, angular rate getters for each axis that return units::degrees_per_second_t, and a Rotation3d getter (pure virtual) that would be constructed from the gyro's internal quaternion representation.